src/test/java/stacks/StacksIT.java

```diff
@@ -0,0 +1,143 @@
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+
+import io.github.bonigarcia.wdm.WebDriverManager;
+import java.util.List;
+
+public class StacksIT {
+    private static WebDriver driver;
+    private static String baseUrl = "https://localhost:9443/instance/stacks/";
+    private static JavascriptExecutor js;
+
+    @BeforeClass
+    public static void setupClass() throws IOException {
+        // Manages the browser driver binary for us
+        WebDriverManager.chromedriver().setup();
+        ChromeOptions options = new ChromeOptions();
+        options.addArguments("--ignore-certificate-errors", "--no-sandbox", "--disable-dev-shm-usage", "--disable-gpu");
+        options.setHeadless(true);
+        driver = new ChromeDriver(options);
+        driver.get(baseUrl);
+
+        js = (JavascriptExecutor) driver;
+
+        String stacksJSON = new String(Files.readAllBytes(Paths.get("src", "test", "resources", "stacks.json")),
+                StandardCharsets.UTF_8);
+
+        // execute javascript functions to set mock data
+        js.executeScript("updateStackView(JSON.parse(arguments[0]));", stacksJSON);
+    }
+
+    @AfterClass
+    public static void teardown() {
+        if (driver != null) {
+            driver.quit();
+        }
+    }
+
+    @Test
+    public void hasCorrectTitleIT() {
+        String expectedTitle = "Stacks";
+        String actualTitle = driver.getTitle();
+        assertEquals("has correct title", expectedTitle, actualTitle);
+    }
+
+    @Test
+    public void stacksTableHasCorrectNumberOfHeadersIT() {
+        WebElement table = driver.findElement(By.id("stack-table"));
+        List<WebElement> tableHeaders = table.findElements(By.tagName("th"));
+
+        int expectedNumberOfStackTableHeaders = 4;
+        int actualNumberOfStackTableHeaders = tableHeaders.size();
+
+        assertEquals("has correct number of table headers", expectedNumberOfStackTableHeaders, actualNumberOfStackTableHeaders);
+    }
+
+    @Test
+    public void stacksTableHasCorrectHeadersIT() {
+        WebElement table = driver.findElement(By.id("stack-table"));
+        List<WebElement> tableHeaders = table.findElements(By.tagName("th"));
+
+        String expectedFirstHeader = "Stack";
+        String expectedSecondHeader = "Version";
+        String expectedThirdHeader = "Status";
+
+        String actualFirstHeader = tableHeaders.get(0).getText();
+        String actualSecondHeader = tableHeaders.get(1).getText();
+        String actualThirdHeader = tableHeaders.get(2).getText();
+
+        assertEquals("has correct stacks table header", expectedFirstHeader, actualFirstHeader);
+        assertEquals("has correct stacks table header", expectedSecondHeader, actualSecondHeader);
+        assertEquals("has correct stacks table header", expectedThirdHeader, actualThirdHeader);
```

Comment:
we should change up each string so when it fails its more specific on which one is wrong, lets add first, second, third to the strings, for example, "has correct first header for stack table"